python/test/t_EfficientGlobalOptimization_std.py

```diff
@@ -149,7 +149,7 @@ def ackley(X):
 # print(result.getOutputSample())
 
 # local refinement
-algo2 = ot.NLopt(problem, 'LD_LBFGS')
+algo2 = ot.TNC(problem)
```

Comment:
then it must be reenabled in python/test/CMakeLists.txt unconditionally to nlopt:
```
if (NLopt_FOUND)
  ot_pyinstallcheck_test (NLopt_std)
  ot_pyinstallcheck_test (EfficientGlobalOptimization_std)
endif ()
```

Reply:
Ok. I saw I was using a quite old version of NLopt (2.4.2). Maybe the new version is better? Do you have a feedback on it? I also planed to directly interface lbfgsb as the NLopt interface is (or was) pure crap. But it introduces another dependence, and I only found a Fortran implementation of the algorithm (and don't want to reintroduce a Fortran dependence). Do you have an idea on how to benefit from this algorithm with the same level of quality as in scipy?

Reply:
nlopt still uses the same code from luksan originally written in fortran and translated to c whereas scipy uses the nocedal implementation, in fortran.

There are several c ports around though, like
https://github.com/stephenbeckr/L-BFGS-B-C

Reply:
If you want I can have a go at it.

Reply:
Oooppppsss I made a confusion between lbfgsb and slsqp! The problem Benoit Chaigne found when debuggin ANATOLE was that NLopt does'nt pass the stoping criteria as is to the solver, but instead do strange things leading to convergence troubles.  I don't know if lbfgsb is impacted too, but the problem seems to be at an upper level than the crude algorithms.

Reply:
this probably needs to be reported to nlopt

Reply:
Benoit did it in May 2018. I didn't check if it has been taken into account. But it is a side story wrt kriging.

Is this PR ok?

Reply:
yes